xkcd/pkg/index/index.go

```diff
@@ -0,0 +1,158 @@
+package index
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/vespian/go-exercises/xkcd/pkg/pbuff"
+	"github.com/vespian/go-exercises/xkcd/pkg/types"
+	"github.com/vespian/go-exercises/xkcd/pkg/web"
+)
+
+func serialize(a types.AllStories, t types.OndiskSerialization,
+) (
+	blob []byte,
+	err error,
+) {
+	switch t {
+	case types.JSON:
+		blob, err = json.Marshal(a)
+		if err != nil {
+			return nil, fmt.Errorf("Index JSON marshaling failed: %s", err)
```

Comment:
Or rather - I wouldn't - because it is very obvious from the types what they mean. If the returned vars were "[]byte, []byte, err" then suddenly you have no way of knowing what which []byte means ... and naming them makes more sense